built_value_generator/lib/src/value_source_field.dart

```diff
@@ -59,8 +59,10 @@ abstract class ValueSourceField
       .any((metadata) => metadataToStringValue(metadata) == 'nullable');
 
   @memoized
-  bool get isIgnored => element.getter.metadata
-      .any((metadata) => metadataToStringValue(metadata) == 'ignored');
+  bool get isCompared => element.getter.metadata.every((metadata) =>
+      getMetadataField(metadata, 'compare') != null
+          ? getMetadataField(metadata, 'compare').toBoolValue()
+          : true);
 
   @memoized
```

Comment:
This is currently checking the `compare` field on every annotation; I think what we want is to pull out the BuiltValueField value once, and check that here.

You can follow the pattern for the BuiltValue annotation:

https://github.com/google/built_value.dart/blob/master/built_value_generator/lib/src/value_source_class.dart#L37

Reply:
You're right. That looks much better. I'll push the updated code in a bit. 👍